test/TestAssets/OutOfProcDataCollector/SampleDataCollector.cs

```diff
@@ -43,14 +44,14 @@ private void Events_TestCaseStart(object sender, TestCaseStartEventArgs e)
         {
             this.logger.LogWarning(this.context.SessionDataCollectionContext, "TestCaseStarted " + e.TestCaseName);
             this.logger.LogWarning(this.context.SessionDataCollectionContext, "TestCaseStarted " + e.TestElement.FullyQualifiedName);
-            var filename = Path.Combine(AppContext.BaseDirectory, "testcasefilename" + i++ + ".txt");
+            var filename = Path.Combine(FileUtility.GetAppDomainBaseDir(), "testcasefilename" + i++ + ".txt");
```

Comment:
A test asset is now dependent on a test utility. It will pull all unnecessary dependencies from the product/test code.